### PR TITLE
Dynamic cropping

### PR DESCRIPTION
**Pull request by anhtu293**
**Dynamic Cropping**
- Possibility to crop an image to smaller fixed size image in the position we want. The crop position can be parsed by argument `center` which can be `float` or `int`.
- If crop is out of image border, an error is triggered.

### CODE REVIEW

**Review by thibo73800 — CHANGES_REQUESTED** [review]

File: alodataset/transforms.py [hunk 1]
```diff
@@ -858,3 +858,43 @@ class RandomDownScaleCrop(Compose):
...
+    """ Crop image to target crop size at chosen position.
+    """
+    def __init__(self, crop_size, *args, **kwargs):
+        assert all([isinstance(s, int) for s in crop_size])
+        self.crop_size = crop_size
+
+        super().__init__(*args, **kwargs)
+
+    def sample_params(self):
+        return tuple()
```
[the remark this answers is not in the document]
Comment (anhtu293) [reply alodataset/transforms.py]: Yes it's not an issue in my case but I just added also the `sample_params` et `set_params` as in other classes like Rotate and Resize.

File: alodataset/transforms.py [hunk 2]
```diff
@@ -858,3 +858,43 @@ class RandomDownScaleCrop(Compose):
...
+            center_x = center[0]
+            center_y = center[1]
+
+        crop_h, crop_w = self.crop_size[0], self.crop_size[1]
+        left = center_x - crop_w // 2
+        top = center_y - crop_h // 2
+        right = center_x + crop_w // 2
+        bot = center_y + crop_h // 2
+
+        if left < 0 or top < 0 or right > (frame.W - 1) or bot > (frame.H - 1):
```
Comment (Data-Iab) [inline alodataset/transforms.py:896]: I don't know if that's gonna be useful or not but ValueError raises when W or H is the same as one of the corresponding crop sizes (i.e if we want to keep the same frame's H or W and crop the other dim)
Exemple :

```python
center = [0.5, 0.5]
frame.W = 50
frame.H = 20
crop_size = [20, 20]
```

```python
ValueError: Crop coordinates out of image border. Image size: (50, 20), Crop coordinate (top, left, bot, right): (0, 15, 20, 35)
```

Comment (Data-Iab) [inline alodataset/transforms.py:896]: I don't know if that's gonna be useful or not but ValueError raises when W or H is the same as one of the corresponding crop sizes (i.e if we want to keep the same frame's H or W and crop the other dim)
Exemple :

```python
center = [0.5, 0.5]
frame.W = 50
frame.H = 20
crop_size = [20, 20]
```

```python
ValueError: Crop coordinates out of image border. Image size: (50, 20), Crop coordinate (top, left, bot, right): (0, 15, 20, 35)
```

Comment (anhtu293) [reply alodataset/transforms.py:896]: You're right. It's an error in my coordinate calculation. fix commit will be pushed quickly

File: alodataset/transforms.py [hunk 3]
```diff
@@ -881,8 +881,8 @@ def apply(self, frame: Frame, center: Union[Tuple[int, int], Tuple[float, float]
                 Default: (0.5, 0.5)
         """
         if isinstance(center[0], float):
-            center_x = int(frame.W * center[0])
-            center_y = int(frame.H * center[1])
+            center_x = int(frame.W * center[0]) - 1
```
Comment (Data-Iab) [inline alodataset/transforms.py:884]: I don't think that shifting the center is good choice. with the same example above you still have the same problem.
```python
ValueError: Crop coordinates out of image border. Image size: (50, 20), Crop coordinate (top, left, bot, right): (-1, 14, 19, 34)
```

**Review by thibo73800 — APPROVED** [review]

**Comment by thibo73800** [comment]
Will merge once @Data-Iab approved

**Review by Data-Iab — APPROVED** [review]
I think it's all good

**Pull request merged**